File: IBlogmart-SPA/src/app/_models/subcategory.ts
import { Category } from './category';

export interface Subcategory {
    id?: number;
    name: string;
    active: boolean;
    categoryId: number;
    category?: Category;
    mainImageUrl?: string;
    images?: [ {
        id: number;
        url: string;
        isMain: boolean;
        type?: number;
    }];
   }

File: IBlogmart.api/Models/SubCategory.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace IBlogmart.api.Models
{
    public class SubCategory
    {
        [Key]
        public int Id { get; set; }
        public string Name {get;set;}
        public bool Active {get;set;}
        public Category Category {get;set;}
        public int CategoryId {get; set;}
        public ICollection<Image> Images {get;set;}
    }
}